### ISSUE TITLE
SoftDTWLoss Training Calculation

### ISSUE THREAD

**Issue by cvarandia** [mblondel/soft-dtw#23]
Hello. Just wanted to clarify.

Is the SoftDTWLoss function designed to accept arrays of time-series? Since during training and testing in Chainer, the loss is computed per batch (true and predicted), the input to the SoftDTWLoss is a batch of time-series labels and predictions. Does the SoftDTWLoss function consider the whole batch as 1 pair of time series or does it treat each pair of time-series (true and predicted) in a batch pair-wise?

Thank you.

**Comment by mblondel** [comment]
Hi! As you can see [here](https://github.com/mblondel/soft-dtw/blob/master/examples/plot_chainer_MLP.py#L67), I'm looping over the batch in the example code. So `SoftDTWLoss` is between two time series, not between two batches of time series. I think back then I made this choice because the batch has to be a Python list (since each time series can have a different length).

**Comment by cvarandia** [comment]
So in order to get the loss of the whole batch, I need to iterate over all the sequences in the batch. And for each iteration, I need to reshape each sequence to (-1,1)  first before getting the SoftDTWLoss for each pair of sequences to be compared? Thank you for clarifying.

**Comment by mblondel** [comment]
`SoftDTWLoss` expects time series of shape `(length, n_dimensions)`, so if your time series are one dimensional and have shape `(length,)`, you indeed need a reshape.

**Comment by cvarandia** [comment]
![received_644121229504668](https://user-images.githubusercontent.com/65875423/82810104-377bef80-9ec0-11ea-86a3-6b1fdb3ad069.png)

I am new to Chainer and I'm using a manual training loop from a sample code. May I know if the placement of the loss calculation (before loss.backward()) is correct?
